pipeline/live_pipeline.py

```diff
@@ -85,23 +82,37 @@ def upload_log_to_s3(aws_client: client) -> None:
 def handler(event: dict, context) -> None:
     """Main function to run the live pipeline on AWS Lambda"""
     nltk.data.path.append("./tmp")
+    logger = initialise_logger()
     aws_client = get_client()
     log_json = read_from_json(aws_client)
     log_date, log = extract_log_and_date(log_json)
     live_url = construct_live_url(log_date)
 
     max_page_num = get_max_page_num(live_url)
     if max_page_num == 0:
-        return "No new data to insert, exiting"
+        logger.info("No new data to insert, exiting")
+        return None
 
-    process_pages(live_url, max_page_num, log)
+    count = 0
+    for page_num in range(1, max_page_num + 1):
+        gpt_response = []
+        data = get_listing_data(live_url, page_num, log)
+        count += len(data)
+        for index, _ in enumerate(data):
```

Comment:
What is the "_" referencing here?

Reply:
It's an empty placeholder which is typical convention for when you don't need a variable. Of course I could've just use range(len) but enumerate was kept to retain consistency with batch_pipeline.